rust/worker/src/execution/operators/metadata_filtering.rs

```diff
@@ -0,0 +1,1348 @@
+                                            if ((*float_value) as f32) <= (*where_value) {
+                                                result.insert(*offset_id);
+                                            }
+                                        }
+                                    }
+                                    _ => (),
+                                }
+                            }
+                        }
+                        return result;
+                    }
+                    WhereClauseComparator::GreaterThan => {
+                        let mut result = RoaringBitmap::new();
+                        // Construct a bitmap consisting of all offset ids
+                        // that have this key > this value.
+                        for (offset_id, meta_map) in &ids_to_metadata {
+                            if let Some(val) = meta_map.get(&metadata_key.to_string()) {
+                                match *val {
+                                    MetadataValue::Float(float_value) => {
+                                        if let KeyWrapper::Float32(where_value) = metadata_value {
+                                            if ((*float_value) as f32) > (*where_value) {
+                                                result.insert(*offset_id);
+                                            }
+                                        }
+                                    }
+                                    _ => (),
+                                }
+                            }
+                        }
+                        return result;
+                    }
+                    WhereClauseComparator::GreaterThanOrEqual => {
+                        let mut result = RoaringBitmap::new();
+                        // Construct a bitmap consisting of all offset ids
+                        // that have this key >= this value.
+                        for (offset_id, meta_map) in &ids_to_metadata {
+                            if let Some(val) = meta_map.get(&metadata_key.to_string()) {
+                                match *val {
+                                    MetadataValue::Float(float_value) => {
+                                        if let KeyWrapper::Float32(where_value) = metadata_value {
+                                            if ((*float_value) as f32) >= (*where_value) {
+                                                result.insert(*offset_id);
+                                            }
+                                        }
+                                    }
+                                    _ => (),
+                                }
+                            }
+                        }
+                        return result;
+                    }
+                },
+                crate::types::MetadataType::StringListType => {
+                    todo!();
```

Comment:
do we intend to do this in a future PR?

nit: prefer if we just use crate::types::*; So this doesn't need to be a verbose import

Reply:
Oh, I was not aware that you are not aware that Ben isn't implementing these in the first cut. I've simply copied pasted a lot of this from his code